dashboard/app/models/course.rb

```diff
@@ -145,10 +145,16 @@ def update_scripts(new_scripts, alternate_scripts = nil)
       default_script = Script.find_by_name!(hash['default_script'])
       # alternate scripts should have the same position as the script they replace.
       position = default_course_scripts.find_by(script: default_script).position
-      CourseScript.find_or_create_by!(course: self, script: alternate_script) do |cs|
+      course_script = CourseScript.find_or_create_by!(course: self, script: alternate_script) do |cs|
```

Comment:
You don't need to pass this block to `find_or_create_by!` if you're going to do the same update below, or can use `tap` and get rid of the update:
```
CourseScript.find_or_create_by!(...).tap do |cs|
  ...
  cs.save
end
```

Reply:
Both of these approaches fail because position is a required field.